### PR TITLE
add config option for overriding auth0 audience

### PR DESCRIPTION
**Pull request by vilterp**
I'm using a local txn service and a local rai engine.

Without this, I was getting:

```
$ python3 examples/run_query.py -p local vilterp-test-3 vilterp-test '2+3'
status: 403
{
  "error": "access_denied",
  "error_description": "Service not enabled within domain: https://127.0.0.1/"
}
```

With this `~/.rai/config`:

```
[local]
infra = AZURE
host = 127.0.0.1
port = 8082
scheme = http
region = us-east
client_id = ...
client_secret = ...
client_credentials_url = https://relationalai-ux.us.auth0.com/oauth/token
```

Now, I can spoof auth0 into thinking I'm not on localhost, but the actual site:

```diff
[local]
infra = AZURE
host = 127.0.0.1
port = 8082
scheme = http
region = us-east
client_id = ...
client_secret = ...
client_credentials_url = https://relationalai-ux.us.auth0.com/oauth/token
+ audience = https://azure-ux.relationalai.com
```

Not sure if there's a different way of doing this, or if it's something we want to have in the public SDK 🤔

### CODE REVIEW

**Comment by salamehsameera** [comment]
lol Pete are you a hacker? I love this workaround but yeah not sure since this is a public repo. Also should we check other SDKs if the same issue exists?

**Comment by vilterp** [comment]
Yeah, I'm not sure it exposes a vulnerability since you still have to have created the keys in auth0, but still, it's a bit of a wart to expose this in the public SDK…

Is there some other way of using this SDK with Oauth creds and a local txn service that people have figured out? cc @bradlo

File: railib/api.py [hunk 1]
```diff
@@ -136,11 +136,12 @@ class Permission(str, Enum):
 # Context contains the state required to make rAI API calls.
 class Context(rest.Context):
     def __init__(self, host: str = None, port: str = None, scheme: str = None,
-                 region: str = None, credentials=None):
+                 region: str = None, credentials=None, auth0_audience: str = None):
         super().__init__(region=region, credentials=credentials)
         self.host = host
         self.port = port or "443"
         self.scheme = scheme or "https"
+        self.auth0_audience = auth0_audience
```
Comment (bradlo) [inline railib/api.py:144]: can we just call this "audience" .. i believe this is an OAuth concept and not specific to auth0.

**Review by bradlo — COMMENTED** [review]
i am fine adding to the SDK if we make that minor name change.

**Review by geokollias — APPROVED** [review]
I think it's fine to expose it as an optional value :+1: Let's do `audience` as Brad suggested.

**Comment by salamehsameera** [comment]
Merging this because I really need it and it looks like it's ready. Thanks @vilterp!

**Comment by vilterp** [comment]
Haha nice, thanks for finally merging 🙏

**Pull request merged**